src/main/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImpl.java

```diff
@@ -811,4 +813,44 @@ public boolean deleteTransaction(String transactionId) throws DataException {
             throw new DataException("Error deleting transaction", ex);
         }
     }
+
+    @Override
+    public CompanyData createPublicCompanyData(PublicCompanySpec publicCompanySpec) throws DataException {
+        var companySpec = new CompanySpec();
+
+        // Only set allowed fields from PublicCompanySpec
+        companySpec.setCompanyNumber(publicCompanySpec.getCompanyNumber());
+        companySpec.setJurisdiction(publicCompanySpec.getJurisdiction());
+        companySpec.setCompanyType(publicCompanySpec.getCompanyType());
+        companySpec.setCompanyStatus(publicCompanySpec.getCompanyStatus());
+        companySpec.setSubType(publicCompanySpec.getSubType());
+        companySpec.setHasSuperSecurePscs(publicCompanySpec.getHasSuperSecurePscs());
+        companySpec.setNumberOfAppointments(publicCompanySpec.getNumberOfAppointments());
+        companySpec.setSecureOfficer(publicCompanySpec.getSecureOfficer());
+        companySpec.setNoDefaultOfficer(publicCompanySpec.getNoDefaultOfficer());
+        companySpec.setRegisters(publicCompanySpec.getRegisters());
+        companySpec.setCompanyStatusDetail(publicCompanySpec.getCompanyStatusDetail());
+        companySpec.setFilingHistoryList(publicCompanySpec.getFilingHistoryList());
+        if (publicCompanySpec.getFilingHistoryList() != null) {
+            companySpec.getFilingHistoryList().forEach(filing -> filing.setDocumentMetadata(false));
+        }
+        companySpec.setNumberOfAppointments(publicCompanySpec.getNumberOfAppointments());
+        companySpec.setOfficerRoles(publicCompanySpec.getOfficerRoles());
+        companySpec.setAccountsDueStatus(publicCompanySpec.getAccountsDueStatus());
+        companySpec.setNumberOfPsc(publicCompanySpec.getNumberOfPsc());
+        companySpec.setPscType(publicCompanySpec.getPscType());
+        companySpec.setPscActive(publicCompanySpec.getPscActive());
+        companySpec.setWithdrawnStatements(publicCompanySpec.getWithdrawnStatements());
+        companySpec.setActiveStatements(publicCompanySpec.getActiveStatements());
+        companySpec.setHasUkEstablishment(publicCompanySpec.getHasUkEstablishment());
+        companySpec.setRegisteredOfficeIsInDispute(publicCompanySpec.getRegisteredOfficeIsInDispute());
+        companySpec.setUndeliverableRegisteredOfficeAddress(
+                publicCompanySpec.getUndeliverableRegisteredOfficeAddress());
+        if (publicCompanySpec.getForeignCompanyLegalForm() != null && publicCompanySpec.getForeignCompanyLegalForm()) {
+            companySpec.setForeignCompanyLegalForm("legal form for company "
```

Comment:
where did setForeignCompanyLegalForm come from - it's not on the original message